src/main/java/org/folio/inventory/validation/experimental/AbstractTargetItemStatusValidator.java

```diff
@@ -0,0 +1,44 @@
+package org.folio.inventory.validation.experimental;
+
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemStatusName;
+import org.folio.inventory.exceptions.UnprocessableEntityException;
+import org.folio.inventory.support.http.server.ValidationError;
+
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static org.folio.inventory.support.CompletableFutures.failedFuture;
+
+public abstract class AbstractTargetItemStatusValidator {
+  private ItemStatusName itemStatusName;
+  private Set<ItemStatusName> allowedStatusToMark;
+
+  protected AbstractTargetItemStatusValidator(ItemStatusName itemStatusName, Set<ItemStatusName> allowedStatusToMark) {
```

Comment:
As this parameter represents a set of statuses I think it should have a plural name. Given that I'm suggesting it be changed, I suggest we use the term `source` to act as a complimentary term to `target`.

```suggestion
  protected AbstractTargetItemStatusValidator(ItemStatusName itemStatusName, Set<ItemStatusName> allowedSourceStatuses) {
```